internal/controller/controllers/installenv_controller_test.go

```diff
@@ -146,6 +147,86 @@ var _ = Describe("installEnv reconcile", func() {
 		Expect(conditionsv1.FindStatusCondition(installEnvImage.Status.Conditions, adiiov1alpha1.ImageCreatedCondition).Status).To(Equal(corev1.ConditionFalse))
 	})
 
+	It("create new installEnv image - cluster not retrieved from database", func() {
+		clusterDeployment := newClusterDeployment("clusterDeployment", testNamespace, getDefaultClusterDeploymentSpec("clusterDeployment-test", "pull-secret"))
+		Expect(c.Create(ctx, clusterDeployment)).To(BeNil())
+
+		expectedError := common.NewApiError(http.StatusInternalServerError, errors.New("server error"))
+		mockInstallerInternal.EXPECT().GetClusterByKubeKey(gomock.Any()).Return(nil, expectedError)
+		installEnvImage := newInstallEnvImage("installEnvImage", testNamespace, adiiov1alpha1.InstallEnvSpec{
+			ClusterRef: &adiiov1alpha1.ClusterReference{Name: "clusterDeployment", Namespace: testNamespace},
+		})
+		Expect(c.Create(ctx, installEnvImage)).To(BeNil())
+
+		res, err := ir.Reconcile(newInstallEnvRequest(installEnvImage))
```

Comment:
Do we want to requeue on Internal Server Error?

Reply:
if I'm not mistaken, we're not supposed to requeue for generic errors (but we do for resource `x` not found, because the could change in the next reconcile). 

tagging @filanov, who might know about other considerations.

Reply:
usually internal error is unexpected like disconnection from the DB/S3 or any other external operation that we can probably retry. 
So if we have 5xx errors then user can not do anything about it, so the controller should try the operation again. 
When we have 4xx errors users probably need to fix it, so we don't need to requeue on 4xx errors.
409 is an exception.